aten/src/ATen/native/LinearAlgebra.cpp

```diff
@@ -126,7 +126,7 @@ Tensor matrix_rank(const Tensor& self, double tol, bool symmetric) {
            "of floating types");
 
   Tensor S = _matrix_rank_helper(self, symmetric);
-  return (S > tol).sum();
+  return (S > tol).to(at::kByte).sum();
```

Comment:
`torch.sum` takes a desired dtype, right? Would be better to provide that instead, which saves you from having to do this conversion, and then do another promotion (I am assuming there's a promotion going on here, but I haven't checked...)

Reply:
I'm going to call this Note "Sum into dtype" and just reference this in other locations I see it.